### PR TITLE
MapView reuse, integration test suite

### PR DESCRIPTION
**Pull request by tobrun**
Follow up from https://github.com/mapbox/mapbox-gl-native/pull/13926 and https://github.com/mapbox/mapbox-gl-native/pull/13976. Since 7.2.0, our map freezes when used in a View reuse scenario. Since this went unnoticed, I'm am adding a new integration test suite using UiAutomator. 

#### MapView freezes in RecyclerView - https://github.com/mapbox/mapbox-gl-native/commit/4d98991e46fffea79f05fbdb634ba0135fac2750

 - GLSurfaceView, calling `OnSurfaceCreated` in a similar fashion to `OnSurfaceDestroyed`. 
 - TextureView, not destroying the map renderer as part of surface destruction, surface can be recreated

#### Integration test suite - https://github.com/mapbox/mapbox-gl-native/commit/676d7dd5c02f4da4727ae2d385a6c86dbcfbddbc
This PR introduces an integration test-suite that will harden against regressions. I added a handful of others occurrences that had regressed in the past. 

image

cc @mapbox/maps-android

### CODE REVIEW

**Comment by tobrun** [comment]
@ivovandongen 
Both master as before https://github.com/mapbox/mapbox-gl-native/pull/13926, we do not allow reuse of a MapView using a GLSurfaceView. 

Below a gif showing the crash:




The crash occurs when you scroll the MapView back into view. When that happens a  `nativeOnSurfaceCreated` is triggered from the render thread and execution fails on `renderer.reset()` in [map_renderer#L169](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/src/map_renderer.cpp#L169). 

The symbolized stack trace [here](https://gist.github.com/tobrun/b2359ed1ed36f8f50683201c4f92e90d).

**Comment by ivovandongen** [comment]
@tobrun Like we discussed on chat. I think ideally we would introduce finer grained callbacks, distinguishing from (EGL) context and surface creation/destroys. Cleanup of the renderer /  backend can then happen on the context related signals and no longer on surface signals so that re-use of the context with a different surface can be supported seamlessly.

In the TextureViewRenderer we have plenty of control and it should be relatively easy to do. For the GLSurfaceViewRenderer, I think we can set a custom `EGLContextFactory` that can provide us the needed signals.

Let's make that a new issue though and continue on this path now.

**Comment by ivovandongen** [comment]
BTW: great addition of the integration tests! Makes everything so much easier to reproduce.

File: platform/android/MapboxGLAndroidSDKTestApp/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -173,10 +173,11 @@
         <activity
                 android:name=".activity.fragment.FragmentBackStackActivity"
                 android:description="@string/description_map_fragment_backstack"
-                android:label="@string/activity_map_fragment_backstack">
+                android:label="@string/activity_map_fragment_backstack"
+                android:launchMode="singleTask">
```
[the remark this answers is not in the document]
Comment (tobrun) [reply platform/android/MapboxGLAndroidSDKTestApp/src/main/AndroidManifest.xml]: verified, nice find

**Pull request merged**